tests/Benchmarks/Http/HttpParser.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Http.Parser;
+
+public class HttpParser
+{
+    private const string _plaintextTechEmpowerRequest =
+        "GET /plaintext HTTP/1.1\r\n" +
+        "Host: localhost\r\n" +
+        "Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7\r\n" +
+        "Connection: keep-alive\r\n" +
+        "\r\n";
+
+    private const string _plaintextTechEmpowerHeaders =
+    "Host: localhost\r\n" +
+    "Accept: text/plain,text/html;q=0.9,application/xhtml+xml;q=0.9,application/xml;q=0.8,*/*;q=0.7\r\n" +
+    "Connection: keep-alive\r\n" +
+    "\r\n";
+
+    private static readonly byte[] s_plaintextTechEmpowerHeadersArray = Encoding.UTF8.GetBytes(_plaintextTechEmpowerHeaders);
+    private static readonly byte[] s_plaintextTechEmpowerRequestArray = Encoding.UTF8.GetBytes(_plaintextTechEmpowerRequest);
+
+    private static readonly ReadOnlySequence<byte> s_plaintextTechEmpowerRequestRos = new ReadOnlySequence<byte>(s_plaintextTechEmpowerRequestArray);
+    private static readonly ReadOnlySequence<byte> s_plaintextTechEmpowerHeadersRos = new ReadOnlySequence<byte>(s_plaintextTechEmpowerHeadersArray);
+
+    private static readonly System.Text.Http.Parser.HttpParser s_parser = new System.Text.Http.Parser.HttpParser();
+
+    [Benchmark]
+    public void RequestLine()
+    {
+        var request = new Request();
```

Comment:
@KrzysztofCwalina do parsing methods have any side effects on `Request`? if not, maybe we should make it a field too?

Reply:
The code with side effects is commented out. After removing the comments the request cannot be a filed. Also, the cost of creating the request object is not avoidable, so I think it's best include it in the cost.